pkg/operator/collection.go

```diff
@@ -523,7 +523,7 @@ func (r *collectionReconciler) makeCollectorConfig(ctx context.Context, spec *mo
 			Type:   monitoringv1.ConfigurationCreateSuccess,
 			Status: corev1.ConditionTrue,
 		}
-		cfgs, err := pmon.ScrapeConfigs()
+		cfgs, err := pmon.ScrapeConfigs(r.opts.ProjectID, r.opts.Location, r.opts.Cluster)
```

Comment:
So IIUC we want to prefer the `OperatorConfig.Collection.ExternalLabels` over operator args, so we want something like:
```go
var projectID = r.opts.ProjectID
if p, ok := spec.ExternalLabels["project_id"]; ok {
	projectID = p
}
...
// etc for location, cluster
cfgs, err := pmon.ScrapeConfigs(projectID, location, cluster)
```

Reply:
Yes, that'd also go for `OperatorConfig.Rules.ExternalLabels` when generating the rule files.

That the `--export.label.*` flag, which we set from the operator's flags, will be overriden by the external labels is already ensured by the precedence change from our last release. So it only needs to be considered in the generated rule and scrape configs.

That's worth doing in a follow up I think.

Reply:
follow-up in https://github.com/GoogleCloudPlatform/prometheus-engine/pull/248